### PR TITLE
WordPress.com Toolbar: Add Site Monitoring menu item

### PR DESCRIPTION
**Pull request by katinthehatsite**
Fixes #https://github.com/Automattic/dotcom-forge/issues/3391
Related to: https://github.com/Automattic/wpcomsh/pull/1464

## Proposed changes:
* Jetpack plugin can add a new `Site Monitoring` menu item to the Tools menu
* Jetpack plugin removes the `Site Logs` menu item

Screenshot 2023-08-14 at 3 45 11 PM

At the moment, the `Site Monitoring` menu item is added through WPCOMSH:  https://github.com/Automattic/wpcomsh/pull/146 . We would like to add it through Jetpack to be consistent with other menu items and remove it from WPCOMSH. 

**Additional Notes**

* Once this PR is deployed and the `Site Monitoring` menu item is added with Jetpack, we will remove the code from WPCOMSH (at the moment, the code makes a conditional check to see if the menu item is added by Jetpack and if not, it adds it. If yes, it skips adding it). 

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

**To test on Atomic sites**
* Navigate to your Atomic Site on WordPress.com
* Add the [Jetpack Beta plugin](https://jetpack.com/download-jetpack-beta/)
* Once the plugin is active, go to `Jetpack > Jetpack Beta > Jetpack` and enable the `add/site-monitoring-menu-item-for-atomic` branch.
* Navigate to **Tools** menu item and confirm that **Site Monitoring** appears under **Tools** once and **is not** added a second time

**To check tests**:

* Pull the changes from this branch locally
* Run `jetpack docker phpunit -- --filter=Test_Atomic_Admin_Menu` in your terminal to confirm that all the tests are passing
* The expected outcome from running tests in that file is:

Screenshot 2023-08-18 at 9 56 30 AM

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 1]
```diff
@@ -388,17 +388,17 @@ public function add_tools_menu() {
...
 		 * @module masterbar
 		 *
 		 * @since 12.0
 		 *
-		 * @param bool $show_wpcom_site_logs_menu Load the WordPress.com Site Logs submenu item. Default to false.
+		 * @param bool $show_wpcom_site_monitoring_menu Load the WordPress.com Site Monitoring submenu item. Default to false.
 		 */
-		if ( apply_filters( 'jetpack_show_wpcom_site_logs_menu', false ) ) {
-			add_submenu_page( 'tools.php', esc_attr__( 'Site Logs', 'jetpack' ), __( 'Site Logs', 'jetpack' ), 'manage_options', 'https://wordpress.com/site-logs/' . $this->domain, null, 7 );
+		if ( apply_filters( 'jetpack_show_wpcom_site_monitoring_menu', false ) ) {
```
Comment (danielbachhuber) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: If we merge this PR _after_ we launch the feature via wpcomsh, then this filter isn't necessary.

**Review by danielbachhuber — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php (hunk 1 above)
Comment (wojtekn) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Having this filter leaves a way to enable/disable that link on `wpcomsh` side, for example, if we would like to enable it only for particular A12s. Do we assume that at the point of Jetpack release, we want to enable it for all users?

Comment (danielbachhuber) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: > Do we assume that at the point of Jetpack release, we want to enable it for all users?

@wojtekn Yes. We're launching the feature with wpcomsh. We'll merge the Jetpack PR afterwards, and then remove the wpcomsh code once Jetpack is deployed to Atomic.

**Review by jeherve — DISMISSED** [review]
LGTM!

**Review by danielbachhuber — APPROVED** [review]
Works as expected 👍 

I created a new Business site, marked it a WoA dev site, installed the Jetpack Beta Tester plugin, checked out this branch, and verified the menu item still appeared.

I then spun up https://github.com/Automattic/wpcomsh/pull/1475, synced it to the site, and verified the menu item still appeared.

**Comment by danielbachhuber** [comment]
WordPress.com test failure is unrelated.

**Pull request merged**